draft-ietf-quic-http.md

```diff
@@ -585,6 +585,116 @@ The payload consists of:
+SHOULD be used.  However, a GOAWAY MAY still be sent to provide additional
+detail to clients.
+
+~~~~~~~~~~  drawing
+ +---------------------------------------------------------------+
+ |                   Last-Stream-ID (32)                         |
+ +---------------------------------------------------------------+
+ |                     [Error Code (32)]                         |
+ +---------------------------------------------------------------+
+ |                 [Additional Debug Data (*)]                 ...
+ +---------------------------------------------------------------+
+~~~~~~~~~~
+ {: #fig-goaway title="GOAWAY frame payload"}
+
+The GOAWAY frame does not define any flags.
+
+The GOAWAY frame applies to the connection, not a specific stream. An endpoint
+MUST treat a GOAWAY frame on a stream other than the control stream as a
+connection error ({{errors}}) of type HTTP_WRONG_STREAM.
+
+The last stream identifier in the GOAWAY frame contains the highest-numbered
+stream identifier for which the sender of the GOAWAY frame might have taken some
+action on or might yet take action on.  All streams up to and including the
+identified stream might have been processed in some way.  The last stream
+identifier can be set to 0 if no streams were processed.
+
+> Note: In this context, "processed" means that some data from the stream was
+> passed to some higher layer of software that might have taken some action as a
+> result.
+
+If a connection terminates without a GOAWAY frame, the last stream identifier is
+effectively the highest possible stream identifier.
+
+Reattempting requests on streams with lower- or equal-numbered identifiers is
+not possible, with the exception of idempotent actions like HTTP GET, PUT, or
+DELETE.  Any protocol activity that used higher-numbered streams can be safely
+retried using a new connection.
+
+Activity on streams numbered lower or equal to the last stream identifier might
+still complete successfully.  The sender of a GOAWAY frame might gracefully shut
+down a connection by sending a GOAWAY frame, maintaining the connection in an
+"open" state until all in-progress streams complete.
+
+An endpoint MAY send multiple GOAWAY frames if circumstances change. For
+instance, an endpoint that sends GOAWAY without an error code during graceful
+shutdown could subsequently encounter an error condition.  The last stream
+identifier from the last GOAWAY frame received indicates which streams could
+have been acted upon.  Endpoints MUST NOT increase the value they send in the
+last stream identifier, since the peers might already have retried unprocessed
+requests on another connection.
+
+A client that is unable to retry requests loses all requests that are in flight
+when the server closes the connection.  A server that is attempting to
+gracefully shut down a connection SHOULD send an initial GOAWAY frame with the
+last stream identifier set to 2^32-1.  This signals to the client that a
```

Comment:
This requirement is no longer correct because we have MAX_STREAM_ID to govern this.  You can send a GOAWAY with the value of MAX_STREAM_ID and be reasonably sure that all in-flight requests are covered.